src/constants/shaders/volumePick.frag

```diff
@@ -0,0 +1,241 @@
+  //   filtering, but three makes this difficult to do through a WebGLRenderTarget.
+  loc0 = floor(loc0 * textureRes) / textureRes;
+  loc0 += vec2(0.5) / textureRes;
+
+  float z = min(floor(pos.z * nSlices), nSlices-1.0);
+  
+  if (flipVolume.z == -1.0) {
+    z = nSlices - z - 1.0;
+  }
+
+  vec2 o = offsetFrontBack(z) + loc0;
+  uint voxelColor = texture2D(tex, o).x;
+
+  // Apply mask
+//   float voxelMask = texture2D(textureAtlasMask, o).x;
+//   voxelMask = mix(voxelMask, 1.0, maskAlpha);
+//   voxelColor.rgb *= voxelMask;
+
+  return bounds*voxelColor;
+}
+
+bool intersectBox(in vec3 r_o, in vec3 r_d, in vec3 boxMin, in vec3 boxMax,
+                  out float tnear, out float tfar) {
+  // compute intersection of ray with all six bbox planes
+  vec3 invR = vec3(1.0,1.0,1.0) / r_d;
+  vec3 tbot = invR * (boxMin - r_o);
+  vec3 ttop = invR * (boxMax - r_o);
+
+  // re-order intersections to find smallest and largest on each axis
+  vec3 tmin = min(ttop, tbot);
+  vec3 tmax = max(ttop, tbot);
+
+  // find the largest tmin and the smallest tmax
+  float largest_tmin  = max(max(tmin.x, tmin.y), tmin.z);
+  float smallest_tmax = min(min(tmax.x, tmax.y), tmax.z);
+
+  tnear = largest_tmin;
+  tfar = smallest_tmax;
+
+  // use >= here?
+  return(smallest_tmax > largest_tmin);
+}
+
+vec4 integrateVolume(vec4 eye_o,vec4 eye_d,
+                     float tnear,   float tfar,
+                     float clipNear, float clipFar,
+                     usampler2D textureAtlas
+                     ) {
+  uint C = 0u;
+  // march along ray from front to back, accumulating color
+
+  // estimate step length
+  const int maxSteps = 512;
+  // modify the 3 components of eye_d by volume scale
+  float scaledSteps = float(BREAK_STEPS) * length((eye_d.xyz/volumeScale));
+  float csteps = clamp(float(scaledSteps), 1.0, float(maxSteps));
+  float invstep = (tfar-tnear)/csteps;
+  // special-casing the single slice to remove the random ray dither.
+  // this removes a Moire pattern visible in single slice images, which we want to view as 2D images as best we can.
+  float r = (SLICES==1.0) ? 0.0 : rand(eye_d.xy);
+  // if ortho and clipped, make step size smaller so we still get same number of steps
+  float tstep = invstep*orthoThickness;
+  float tfarsurf = r*tstep;
+  float overflow = mod((tfarsurf - tfar),tstep); // random dithering offset
+  float t = tnear + overflow;
+  t += r*tstep; // random dithering offset
+  float tdist = 0.0;
+  int numSteps = 0;
+  vec4 pos, col;
+  for (int i = 0; i < maxSteps; i++) {
+    pos = eye_o + eye_d*t;
+    // !!! assume box bounds are -0.5 .. 0.5.  pos = (pos-min)/(max-min)
+    // scaling is handled by model transform and already accounted for before we get here.
+    // AABB clip is independent of this and is only used to determine tnear and tfar.
+    pos.xyz = (pos.xyz-(-0.5))/((0.5)-(-0.5)); //0.5 * (pos + 1.0); // map position from [boxMin, boxMax] to [0, 1] coordinates
```

Comment:
```suggestion
    pos.xyz = pos.xyz + 0.5; // map position from [boxMin, boxMax] to [0, 1] coordinates
```
Equivalent?

Reply:
equivalent yes, but less clear as to exactly why.  I could see making this change but keeping more of the derivation steps in a comment.  Also this needs to match up with the raymarch rendering shader.